### ISSUE TITLE
Redundant proxy is created for multi-spec specifications when expected type is not a spec-interface

### ISSUE THREAD

**Issue by jradlica** [tkaczmarzyk/specification-arg-resolver#162]
See comment section in below code:
`SpecificationFactory.java`
```java
	public Specification<?> createSpecificationDependingOn(ProcessingContext context) {
		List<Specification<Object>> specs = resolveSpec(context);

		if (specs.isEmpty()) {
			return null;
		}

		if (specs.size() == 1) {
			Specification<Object> firstSpecification = specs.iterator().next();

			if (Specification.class == context.getParameterType()) {
				return firstSpecification;
			} else {
				return (Specification<?>) EnhancerUtil.wrapWithIfaceImplementation(context.getParameterType(), firstSpecification);
			}
		}

		Specification<Object> spec = new net.kaczmarzyk.spring.data.jpa.domain.Conjunction<>(specs);

                // !!! We should check `context.getParameterType()` is of type `Specification.class`, if yes we should skip proxy generation.
                // Eg. When the parameter type is: `Specification` or `Conjunction` we generate proxy for above Conjunction of type Specification/Conjunction (it's redundant, we can just return conjunction) 
		return (Specification<?>) EnhancerUtil.wrapWithIfaceImplementation(context.getParameterType(), spec);
	}
```

Refactoring proposition:
```java
	public Specification<?> createSpecificationDependingOn(ProcessingContext context) {
		List<Specification<Object>> specs = resolveSpec(context);

		if (specs.isEmpty()) {
			return null;
		}

		Specification<Object> spec = specs.size() == 1 ? specs.iterator().next() : new net.kaczmarzyk.spring.data.jpa.domain.Conjunction<>(specs);


		if (Specification.class == context.getParameterType()) {
			return spec;
		} else {
			return (Specification<?>) EnhancerUtil.wrapWithIfaceImplementation(context.getParameterType(), spec);
		}
	}
```

We should also write additional tests that (at least) will verify that for conjunction parameter type the proxy is not generated.

**Comment by tkaczmarzyk** [comment]
@jradlica  Regarding refactoring proposition, I think that `if (Specification.class == context.getParameterType()) {` is not enough, and we should use `if (Specification.class == context.getParameterType() || Conjunction.class == context.getParameterType()) {`?

**Comment by tkaczmarzyk** [comment]
Or even more generally, we should just check if argument is assignable from the parameter type -- because someone could use e.g. `Like` as parameter type and in such case we also don't need to create  any proxy

**Comment by jradlica** [comment]
@tkaczmarzyk I agree, as target solution we should do something like: `if(Specification.class.isAssignableFrom(context.getParamtertType())`

**Comment by tkaczmarzyk** [comment]
will be included in v2.15.1
